aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -71,9 +71,10 @@ void replication_pad1d_out_cpu_template(
   int pad_l = paddingSize[0];
   int pad_r = paddingSize[1];
 
-  TORCH_CHECK(input_.numel() > 0
-      && (input_.ndimension() == 2 || input_.ndimension() == 3),
-      "non-empty 2D or 3D (batch mode) tensor expected for input");
+  // allow empty batch size but not other dimensions.
+  TORCH_CHECK((input_.dim() == 2) ||
+              (input_.dim() == 3 && input_.size(1) != 0 && input_.size(2) != 0),
+              "2D or 3D (batch mode) tensor for input, but got: ", input_.sizes());
```

Comment:
You should check dimensions 0,1 here for a 2d tensor, and 1,2 for a 3d tensor, please see my comment from previous review.